src/backends/plonky2/circuits/mainpod.rs

```diff
@@ -252,9 +313,9 @@ impl OperationVerifyGadget {
         let merkle_proof_ok = builder.all(merkle_proof_checks);
 
         // Check output statement
-        let arg1_key = resolved_op_args[0].args[0].clone();
-        let arg2_key = resolved_op_args[1].args[0].clone();
-        let arg3_key = resolved_op_args[2].args[0].clone();
+        let arg1_key = cache.equations[0].lhs.clone();
```

Comment:
For all the operations that build a statement from entries, this was previously anchored key, so the name `arg1_key` was fitting.

But now this can be an anchored key or literal.  For clarity I would suggest renaming all `argX_key` in all `eval_op` methods to `argX_key_or_lit` or something similar.